tests/ashley/test_templatetags_custom_tags.py

```diff
@@ -0,0 +1,68 @@
+"""Test suite for ashley custom tags"""
+
+from django.template import Context, Template
+from django.test import TestCase
+
+from ashley.factories import ForumFactory, LTIContextFactory, TopicFactory, UserFactory
+
+
+class TestIsUserInstructorTag(TestCase):
+    """
+    Integration tests to validate that the filter is_user_instructor
+    can detect if user is instructor of a forum
+    """
+
+    def setUp(self):
+        """
+        Set up needed data for our tests
+        """
+        # Create two LTI Context
```

Comment:
I wouldn't put this in the setUp. It's a lot of things you do here and you will later add more tests, which will not all need exactly this setup... but it will be created nevertheless and slow down tests considerably.
My personal practice is to have a specific method (if the name does not start by test_ it will not be treated as a test by the runner. Make it start by _ for example) that I explicitly call from everytest that needs exactly this setup.
At this stage, you could also just put the code in your test since there's only 1 :wink:

Reply:
ok thank you for the best practice